### DISCUSSION TITLE
plotPDF xOz plane ?

### DISCUSSION THREAD

**Discussion by ThomasChauve** [mtex-toolbox/mtex#1863] [Tips and Tricks]
Hello

I have a simple question is it possible to plot a pole figure in with y-axis outofPlane (i.e. xOz plane) other then rotation the orientation.

```matlab
%% figure
h = Miller(0,0,1,CS{1},'hkl');
figure
setMTEXpref('FontSize',25)
plotx2east
plotPDF(odf_TGI03,h,'antipodal','resolution',5*degree);
```

![image](https://github.com/mtex-toolbox/mtex/assets/9338747/68ffd6e7-d5a0-4370-a5a0-7d0100f53841)

Best 
Thomas

**Comment by lfgmorales** [comment]
Hi Thomas,

I am not sure if that nwill answer your question, but at the moment you have the pole figure labelling for mtex on, so fitst you have to switch this off in the beginning

setMTEXpref('pfAnnotations',@(varargin) 1);


and then annotate your pole figures the way you want after you plot

annotate (zvector,’label’,{‘Y’},’BackgroundColor’,’w’)


Juast change the letters accordingly to what you want in relation to xvecxtor, yvector and zvector

ccheers


Luiz



On 12 Jun 2023, at 12:30, Chauve ***@***.******@***.***>> wrote:



Hello

I have a simple question is it possible to plot a pole figure in with y-axis outofPlane (i.e. xOz plane) other then rotation the orientation.

%% figure
h = Miller(0,0,1,CS{1},'hkl');
figure
setMTEXpref('FontSize',25)
plotx2east
plotPDF(odf_TGI03,h,'antipodal','resolution',5*degree);

[image]<https://user-images.githubusercontent.com/9338747/245111896-68ffd6e7-d5a0-4370-a5a0-7d0100f53841.png>

Best
Thomas

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1863>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN727IHT6QDSLQMVL3LXK3VWBANCNFSM6AAAAAAZDFIQCE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1863", "url": "https://github.com/mtex-toolbox/mtex/discussions/1863", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by ThomasChauve** [reply]
Thank this will help me but I also want to rotate the data. For now I use :

```matlab
rot90=rotation.byAxisAngle(xvector,90*degree);
pf = rotate(odf_TGI03,rot90)
```

This gives me :

![TGI03_xoz](https://github.com/mtex-toolbox/mtex/assets/9338747/b9149bc7-e812-413a-b01c-59269c87f60b)
![TGI03_xoy](https://github.com/mtex-toolbox/mtex/assets/9338747/13a1a04a-188b-4cff-99a2-1d67a036ba2b)

Then if I add you renaming of label it will wok. But to me it is not really efficient way. I just wonder if there is an option somewhere that I miss.

**Comment by lfgmorales** [comment]
or in a more decent (maybe not that elegant) way

annotate([xvector,yvector,-zvector],'label',{'X','Z','Y'},'backgroundcolor','w','MarkerSize',15,'FontSize',20)

cheers

Luiz

On 12 Jun 2023, at 12:30, Chauve ***@***.******@***.***>> wrote:



Hello

I have a simple question is it possible to plot a pole figure in with y-axis outofPlane (i.e. xOz plane) other then rotation the orientation.

%% figure
h = Miller(0,0,1,CS{1},'hkl');
figure
setMTEXpref('FontSize',25)
plotx2east
plotPDF(odf_TGI03,h,'antipodal','resolution',5*degree);

[image]<https://user-images.githubusercontent.com/9338747/245111896-68ffd6e7-d5a0-4370-a5a0-7d0100f53841.png>

Best
Thomas

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1863>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN727IHT6QDSLQMVL3LXK3VWBANCNFSM6AAAAAAZDFIQCE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1863", "url": "https://github.com/mtex-toolbox/mtex/discussions/1863", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by lfgmorales** [comment]
Hi Thomas

I think this is not implemented, but maybe Ralf or Rüdiger have a work around

If you use import wizard you can already define a rotation there (in Euler), but still, there is the rotation like you have

cheers

Luiz



On 12 Jun 2023, at 13:00, Chauve ***@***.******@***.***>> wrote:



Thank this will help me but I also want to rotate the data. For now I use :

rot90=rotation.byAxisAngle(xvector,90*degree);
pf = rotate(odf_TGI03,rot90)

This gives me :

[TGI03_xoz]<https://user-images.githubusercontent.com/9338747/245118343-b9149bc7-e812-413a-b01c-59269c87f60b.png>
[TGI03_xoy]<https://user-images.githubusercontent.com/9338747/245118353-13a1a04a-188b-4cff-99a2-1d67a036ba2b.png>

Then if I add you renaming of label it will wok. But to me it is not really efficient way. I just wonder if there is an option somewhere that I miss.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1863#discussioncomment-6151177>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4W3U4TUAWQSLVOYNTXK3ZE7ANCNFSM6AAAAAAZDFIQCE>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1863#discussioncomment-6151177", "url": "https://github.com/mtex-toolbox/mtex/discussions/1863#discussioncomment-6151177", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by ralfHielscher** [comment]
Hi Thomas,

with MTEX 6.0.beta1 we have introduced the option to arbitrarily align pole figures. The syntax is as follows
```matlab
how2plot = plottingConvention;
how2plot.outOfScreen = yvector;
how2plot.east = zvector
plotPDF(odf, Miller(1,0,0,odf.CS), how2plot)
```
It would be very nice if you could test this new feature.

All the best,
Ralf.

**Reply by ThomasChauve** [reply]
Hi

Thanks for implementing this.

It tried it but I have a bug :

```matlab
CS = crystalSymmetry('6/mmm', [4.51 4.51 7.34], 'X||a*', 'Y||b', 'Z||c', 'mineral', 'Ice_Ih', 'color', 'light blue');

adr='PIL033 axial whole clean.ctf';

ebsd=EBSD.load(adr,CS,'interface','ctf','convertEuler2SpatialReferenceFrame');
%%


h = Miller(0,0,1,CS,'hkl');

plotPDF(ebsd('Ice_Ih').orientations,h,'antipodal','resolution',5*degree,'contourf');

%%
clc
odf = calcDensity(ebsd.orientations('Ice_Ih'))
how2plot = plottingConvention;
how2plot.outOfScreen = yvector;
how2plot.east = zvector
plotPDF(odf, Miller(1,0,0,odf.CS), how2plot)
```
both the plotPDF with orientation or odf return the following error:

```matlab
how2plot = plottingConvention
 
  outOfScreen: (0,1,0)
  north      : (1,0,0)
  east       : (0,0,1)
 
Undefined function 'clim' for input arguments of type 'matlab.graphics.axis.Axes'.

Error in vector3d/smooth (line 101)
    clim(sP(j).ax,colorRange);

Error in vector3d/plot (line 103)
    [varargout{1:nargout}] = v.smooth(varargin{:});

Error in S2Fun/plot (line 47)
      [h(ul,j),ax(ul,j)] = plot(plotNodes{ul},values{ul}(:,j),'pcolor','hold',S2Proj(ul),varargin{:});

Error in S2FunHarmonicSym/plot (line 21)
[varargout{1:nargout}] = sF.plot@S2Fun(sR,sF.s,varargin{:},sF.s.how2plot);

Error in SO3Fun/plotPDF (line 62)
  [~,cax] = plot(pdf,'smooth','doNotDraw','ensureNonNeg',varargin{:});

Error in Qietal_odf_test60 (line 21)
plotPDF(odf, Miller(1,0,0,odf.CS), how2plot)
```

I am using :

MATLAB R2021a
MTEX 6.0 beta1
OS: Ubuntu 18.04

Best
Thomas

**Reply by ralfHielscher** [reply]
Hi Thomas,

could you please replace `clim` by `caxis` in vector3d/smooth (line 101)
This is a Matlab change in more recent versions.

Ralf.

**Reply by ThomasChauve** [reply]
OK then It works like a charm.

Then what should be the minimum Matlab version ? 

By the way I saw that there is a now a link with Neper. I like it.

Thanks for the work.

Best
Thomas

**Reply by ralfHielscher** [reply]
I uploaded new versions which resolve this problem.
All the best,
Ralf.

**Reply by ThomasChauve** [reply]
It works for me. Perfect.
